pkg/controller/import-controller.go

```diff
@@ -44,315 +51,532 @@ const (
+
+// Reconcile the reconcile loop for the CDIConfig object.
+func (r *ImportReconciler) Reconcile(req reconcile.Request) (reconcile.Result, error) {
+	log := r.Log.WithValues("PVC", req.NamespacedName)
+	log.V(1).Info("reconciling Import PVCs")
+
+	// Get the PVC.
+	pvc := &corev1.PersistentVolumeClaim{}
+	if err := r.Client.Get(context.TODO(), req.NamespacedName, pvc); err != nil {
+		if k8serrors.IsNotFound(err) {
+			return reconcile.Result{}, nil
+		}
+		return reconcile.Result{}, err
+	}
+
+	if !shouldReconcilePVC(pvc) {
+		r.Log.V(1).Info("Should not reconcile this PVC", "pvc.annotation.phase.complete", isPVCComplete(pvc),
+			"pvc.annotations.endpoint", checkPVC(pvc, AnnEndpoint), "pvc.annotations.source", checkPVC(pvc, AnnSource))
+		return reconcile.Result{}, nil
 	}
-	anno := map[string]string{}
 
 	// In case this is a request to create a blank disk on a block device, we do not create a pod.
 	// we just mark the DV as successful
 	volumeMode := getVolumeMode(pvc)
-	if volumeMode == v1.PersistentVolumeBlock && pvc.GetAnnotations()[AnnSource] == SourceNone {
-		klog.V(3).Infof("attempting to create blank disk for block mode, this is a no-op, marking pvc with pod-phase succeeded")
-		anno[AnnPodPhase] = string(v1.PodSucceeded)
-		_, err := updatePVC(ic.clientset, pvc, anno, lab)
-		if err != nil {
-			return errors.WithMessage(err, "could not update pvc %q annotation and/or label")
+	if volumeMode == corev1.PersistentVolumeBlock && pvc.GetAnnotations()[AnnSource] == SourceNone {
+		log.V(1).Info("attempting to create blank disk for block mode, this is a no-op, marking pvc with pod-phase succeeded")
+		if pvc.GetAnnotations() == nil {
+			pvc.SetAnnotations(make(map[string]string, 0))
 		}
-		return nil
-	}
-	// find import Pod
-	pod, err := ic.findImportPodFromCache(pvc)
-	if err != nil {
-		return err
+		pvc.GetAnnotations()[AnnPodPhase] = string(corev1.PodSucceeded)
+		if err := r.updatePVC(pvc); err != nil {
+			return reconcile.Result{}, errors.WithMessage(err, fmt.Sprintf("could not update pvc %q annotation and/or label", pvc.Name))
+		}
+		return reconcile.Result{}, nil
 	}
-	pvcKey, err := cache.MetaNamespaceKeyFunc(pvc)
+
+	// See if we have a pod associated with the PVC, we know the PVC has the needed annotations.
+	pod, err := r.findImporterPod(pvc)
 	if err != nil {
-		return err
+		return reconcile.Result{}, err
 	}
 
-	// Pod must be controlled by this PVC
-	if pod != nil && !metav1.IsControlledBy(pod, pvc) {
-		return errors.Errorf("found pod %s/%s not owned by pvc %s/%s", pod.Namespace, pod.Name, pvc.Namespace, pvc.Name)
-	}
+	return r.reconcilePvc(pvc, pod)
+}
 
-	// expectations prevent us from creating multiple pods. An expectation forces
-	// us to observe a pod's creation in the cache.
-	needsSync := ic.podExpectations.SatisfiedExpectations(pvcKey)
+func (r *ImportReconciler) findImporterPod(pvc *corev1.PersistentVolumeClaim) (*corev1.Pod, error) {
+	podName := fmt.Sprintf("%s-%s", common.ImporterPodName, pvc.Name)
```

Comment:
pod name is constructed in this manner in a couple spots so should probably be a function (that takes pvc as param).  Should prob do the same for scratch pvc name.

Reply:
Good suggestion, done.